main.c

```diff
@@ -312,28 +344,11 @@ int main(int argc, const char *argv[])
 	prog = elf_load_img(uk_alloc_get_default(), (void *) img->vbase,
 			    img->len, progname);
 #else /* CONFIG_APPELFLOADER_VFSEXEC */
-#if CONFIG_APPELFLOADER_VFSEXEC_ENVPATH
-	env_path = getenv("PATH");
-	if (env_path) {
-		realpath = locate_exec(path, env_path);
-		if (PTR2ERR(realpath) == -EINVAL) {
-			realpath = NULL;
-		} else if (PTRISERR(realpath) && PTR2ERR(realpath) != -EINVAL) {
-			uk_pr_err("%s: Failed to find executable in envirenment ($PATH): %s (%d)",
-				  progname, strerror(-PTR2ERR(realpath)),
-				  PTR2ERR(realpath));
-			goto out_free_thread;
-		}
-	}
-#endif /* CONFIG_APPELFLOADER_VFSEXEC_ENVPATH */
 	uk_pr_debug("%s: Load executable (%s)...\n",
 		    progname, realpath ? realpath : path);
 	prog = elf_load_vfs(uk_alloc_get_default(),
 			    realpath ? realpath : path, progname);
-#if CONFIG_APPELFLOADER_VFSEXEC_ENVPATH
```

Comment:
I think you may still want to free this regardless of error or not before going to `sleep()`. Maybe this was why it was here to begin with and the memleak has always existed? I think we should add its free before the `uk_sched_thread_add`

Reply:
I think the root of the problem is the infinite loop, which prevents a lot more memory from being freed during the shutdown path. We will get rid of this in the next release, so it should be okay for now.